lib/auth/auth_with_roles_test.go

```diff
@@ -355,3 +355,152 @@ func serverWithAllowRules(t *testing.T, srv *TestAuthServer, allowRules []types.
 		context:    *authContext,
 	}
 }
+
+func TestGetDatabaseServers(t *testing.T) {
```

Comment:
How do these tests relate to changing the way implicit roles are created? (I have no complaints if they're _not_ related - but maybe add a note in the PR description)

Reply:
Updated the description, those tests would fail if the implicit role change isn't made.